### PR TITLE
chore: adding test for runAdAuction

### PR DESCRIPTION
**Pull request by brodrigu**
Adding e2e test for `runAdAuction`

### CODE REVIEW

File: test/e2e/auction/index.test.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+describe('Fledge', () => {
+	describe('runAdAuction', () => {
+		beforeEach(async () => {
+			await page.goto('http://localhost:3000/test/e2e/');
+		});
+
+		it('should error when no parameters sent', async () => {
+			const fledge = await page.evaluate(() => window.fledge);
+			expect(() => fledge.joinAdInterestGroup()).toThrow();
+		});
```
Comment (iamnewton) [inline test/e2e/auction/index.test.js:11]: should this be testing the auction stuff and not the `joinAdInterestGroup` functions

File: test/e2e/mock/dl.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+/* eslint-disable camelcase */
+export const score_ad = (adMetadata, bid, auctionConfig, trustedScoringSignals, browserSignals) => {
+	console.log('scoring bid...');
```
Comment (iamnewton) [inline test/e2e/mock/dl.js:3]: do we need the console statements in here for a mock?

File: test/e2e/mock/bl.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/* eslint-disable camelcase */
+export const generate_bid = (interestGroup, auctionSignals, perBuyerSignals, trustedBiddingSignals, browserSignals) => {
+	console.log('generating bid...');
```
Comment (iamnewton) [inline test/e2e/mock/bl.js:3]: do we need the console statements in here for a mock?

File: test/e2e/auction/index.test.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+		it('should return token when provided minimum required params and valid interest groups', async () => {
+			await page.evaluate(() => window.fledge.joinAdInterestGroup({
+				owner: 'magnite.com',
+				name: 'test-interest',
+				bidding_logic_url: 'http://localhost:3000/test/e2e/mock/bl.js',
+			}, 100000));
```
Comment (iamnewton) [inline test/e2e/auction/index.test.js:18]: This also seems like leftovers from the `joinAdInterestGroup` rather than the auction tests.

**Review by iamnewton — CHANGES_REQUESTED** [review]

File: test/e2e/mock/dl.js (hunk 2 above)
Comment (brodrigu) [reply test/e2e/mock/dl.js:3]: its helpful when loaded in the browser for debugging and doesn't effect the command line. I can comment them out but that seems to provide no benefit and a small hassle when they are needed.

File: test/e2e/auction/index.test.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+						size_id: 123,
+					},
+					per_buyer_signals: {
+						'dsp.com': {
+							content_quality: 230,
+						},
+					},
+				}),
+			);
+			expect(typeof result).toBe('string');
```
Comment (iamnewton) [inline test/e2e/auction/index.test.js:58]:
```suggestion
			expect(result).toBe(expect.any(String));
```

File: test/e2e/mock/bl.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+/* eslint-disable camelcase */
+export const generate_bid = (interestGroup, auctionSignals, perBuyerSignals, trustedBiddingSignals, browserSignals) => {
```
Comment (iamnewton) [inline test/e2e/mock/bl.js:2]:
```suggestion
export const generateBid = (interestGroup, auctionSignals, perBuyerSignals, trustedBiddingSignals, browserSignals) => {
```

File: test/e2e/mock/dl.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
+/* eslint-disable camelcase */
+export const score_ad = (adMetadata, bid, auctionConfig, trustedScoringSignals, browserSignals) => {
```
Comment (iamnewton) [inline test/e2e/mock/dl.js:2]:
```suggestion
export const scoreAd = (adMetadata, bid, auctionConfig, trustedScoringSignals, browserSignals) => {
```

File: test/e2e/mock/dl.js [hunk 8]
```diff
@@ -0,0 +1,5 @@
+/* eslint-disable camelcase */
```
Comment (iamnewton) [inline test/e2e/mock/dl.js:1]:
```suggestion
```

File: test/e2e/mock/bl.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
+/* eslint-disable camelcase */
```
Comment (iamnewton) [inline test/e2e/mock/bl.js:1]:
```suggestion
```

**Review by iamnewton — CHANGES_REQUESTED** [review]

File: test/e2e/auction/index.test.js (hunk 5 above)
Comment (brodrigu) [reply test/e2e/auction/index.test.js:58]: is this functionally different? seems less clear to me.

File: test/e2e/mock/bl.js (hunk 6 above)
Comment (brodrigu) [reply test/e2e/mock/bl.js:2]: the branch I am working on still expects generate_bid.

File: test/e2e/mock/dl.js (hunk 7 above)
Comment (brodrigu) [reply test/e2e/mock/dl.js:2]: current branch (and main still expect score_ad)

File: test/e2e/mock/dl.js (hunk 8 above)
Comment (brodrigu) [reply test/e2e/mock/dl.js:1]: if we don't change the function names I can't remove

File: test/e2e/auction/index.test.js (hunk 5 above)
Comment (iamnewton) [reply test/e2e/auction/index.test.js:58]: yeah, it's not just a `typeof` string comparison.  It checks the constructor.

**Review by iamnewton — APPROVED** [review]

**Pull request merged**